src/test/java/com/smartcar/sdk/SmartcarExceptionV2Test.java

```diff
@@ -0,0 +1,125 @@
+package com.smartcar.sdk;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import javax.json.Json;
+
+import static org.mockito.Matchers.eq;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Test Suite: SmartcarExceptionV2
+ */
+@PrepareForTest({
+    SmartcarExceptionV2.class,
+    Gson.class,
+    JsonObject.class,
+    Response.class,
+    ResponseBody.class,
+})
+@PowerMockIgnore("javax.net.ssl.*")
+
+public class SmartcarExceptionV2Test extends PowerMockTestCase {
+  /**
+   * Test throwing an SmartcarExceptionV2 with a message.
+   */
+  @Test
+  public void testNewExceptionWithDescription() throws Exception {
+    String testMessage = "test message";
+
+    try {
+      throw new SmartcarExceptionV2(testMessage);
+    }
+    catch (SmartcarExceptionV2 ex) {
+      Assert.assertEquals(ex.getDescription(), testMessage);
+    }
+  }
+
+  /**
+   * Test throwing an SmartcarExceptionV2 with a message.
+   */
+  @Test
+  public void testNewExceptionWithRequestAndResponse() throws Exception {
+    String expectedResolution = "expected resolution";
+    String expectedDescription = "expected description";
+    String expectedType = "ACCOUNT_STATE";
+    String expectedCode = "INVALID_CREDENTIALS";
+    String expectedRequestId = "011660dc-8322-4064-a972-53826c8dff9c";
+    int expectedStatusCode = 200;
+
+    String errorString = "{\"type\": \"" + expectedType + "\"," +
+      "\"code\": \"" + expectedCode + "\"," +
+      "\"description\": \"" + expectedDescription + "\"," +
+      "\"docURL\": null," +
+      "\"statusCode\": 200," +
+            "\"resolution\": \"" + expectedResolution + "\"," +
+            "\"detail\": null," +
+      "\"requestId\": \"" + expectedRequestId + "\"}";
```

Comment:
Please use `Json.createObjectBuilder()` here

Reply:
I will try again. I think there was problems with setting the value to `null`

Reply:
ah I see

check this out: https://stackoverflow.com/a/44631692****